changelogs/unreleased/gh-9855-box-ctl-demote-guarantee-demoting.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/replication
+
+* Fixed a bug that allowed the old leader to get re-elected after resigning
+  himself through `box.ctl.demote` (gh-9855).
```

Comment:
let's clarify that this could happen only with `election_mode = 'candidate'`.

Reply:
Fixed.